### PR TITLE
fix: consumers can be non-fifo

### PR DESCRIPTION
**Pull request by amir27111998**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -187,6 +187,11 @@ export interface Topic {
    * the default queue is not specified, an error will be thrown
    */
   separateConsumerGroup?: string;
+  /**
+   * An optional tells that the consumer is fifo or not.
+   * if not provided, the broker will use the topic's isFifo
+   */
+  isConsumerFifo?: boolean;
```
[the remark this answers is not in the document]
Comment (MGB247) [reply src/types.ts:194]: or better yet
```suggestion
  consumerGroup?: ConsumerOptions;
```

[the remark this answers is not in the document]
Comment (amir27111998) [reply src/types.ts]: done.

File: src/emitters/emitter.sqns.ts [hunk 2]
```diff
@@ -717,6 +718,20 @@ export class SqnsEmitter implements IEmitter {
...
+  private getSeparateConsumer(consumer: Pick<Topic, 'separateConsumerGroup' | 'consumerGroup'>): string | undefined {
+    const { separateConsumerGroup, consumerGroup } = consumer;
+    if (separateConsumerGroup && consumerGroup)
+      throw new Error(
+        `separateConsumerGroup and consumerGroup cannot be used together`
+      );
+    return separateConsumerGroup || consumerGroup?.name;
+  }
+
+  private checkIfConsumerIsFifo(topic: Pick<Topic, 'consumerGroup' | 'isFifo'>): boolean {
```
Comment (MGB247) [inline src/emitters/emitter.sqns.ts:730]: no need to pick here as we are passing topic anyways

Comment (MGB247) [inline src/emitters/emitter.sqns.ts:730]:
```suggestion
  private isConsumerFifo(topic: Pick<Topic, 'consumerGroup' | 'isFifo'>): boolean {
```

File: src/types.ts [hunk 3]
```diff
@@ -181,12 +183,22 @@ export interface Topic {
...
+   * the default queue is not specified, an error will be thrown.
    */
   separateConsumerGroup?: string;
+
+  /**
+   * An optional consumer group specification
+   * Use this when consumer configuration is different from the topic
+   * 
+   * When specified, the broker will consume the messages from this
+   * otherwise the broker will subscribe the default queue to this topi
```
Comment (MGB247) [inline src/types.ts:198]:
```suggestion
   * otherwise the broker will subscribe the default queue to this topic
```

File: src/types.ts [hunk 4]
```diff
@@ -181,12 +183,22 @@ export interface Topic {
...
    *
-   * Used to identify the queue from which the consumer 
+   * Used to identify the queue from which the consumer
    * will consume the messages from this topic. When not provided,
-   * the broker will subscribe the default queue to this topic. If 
-   * the default queue is not specified, an error will be thrown
+   * the broker will subscribe the default queue to this topic. If
+   * the default queue is not specified, an error will be thrown.
    */
   separateConsumerGroup?: string;
```
Comment (MGB247) [inline src/types.ts:191]: mark deprecated

File: src/types.ts (hunk 3 above)
Comment (amir27111998) [reply src/types.ts]: done.

File: src/types.ts (hunk 4 above)
Comment (amir27111998) [reply src/types.ts:192]: done.

File: src/emitters/emitter.sqns.ts (hunk 2 above)
Comment (amir27111998) [reply src/emitters/emitter.sqns.ts]: done.

**Pull request merged**